client/checkout/blocks/confirm-upe-payment.js

```diff
@@ -26,7 +26,7 @@ export default async function confirmUPEPayment(
 			payment_method_data: {
 				billing_details: {
 					name,
-					email: billingData.email || '-',
+					email: billingData.email.trim() || '-',
```

Comment:
Given the fallback, is there a way that the email could be undefined? In which case we would have a JS error when we try to trim it.

Maybe good to use a safe approach?

Reply:
This is a very valid point! I am type-checking this value before using `trim`. 809737ca7775d1abbef8ef65435eea5a64c44541